src/instruction/incl.rs

```diff
@@ -0,0 +1,184 @@
+
+        // If a source has already been included, skip it without returning
+        // an error
+        if i.is_included(&formatted) {
+            return Ok((formatted, vec![]));
+        }
+
+        i.debug("FINAL PATH", &format!("{:?}", formatted));
+
+        let input = std::fs::read_to_string(&formatted)?;
+
+        // We can't just parse the input, since it adds the instructions
+        // to an entry block in order to execute them. What we can do, is
+        // parse many instructions and add them to an empty interpreter
+        let (_, instructions) = Construct::many_instructions(input.as_str())?;
+
+        Ok((formatted, instructions))
+    }
+
+    /// Try to load code from the current path where the executable has been launched
+    fn load_relative(
+        &self,
+        base: &Path,
+        i: &Interpreter,
+    ) -> Result<(PathBuf, Vec<Box<dyn Instruction>>), JkError> {
+        self.inner_load(base, i)
+    }
+
+    /// Try to load code from jinko's installation path
+    fn _load_jinko_path(&self) -> Result<Vec<Box<dyn Instruction>>, JkError> {
+        todo!()
+    }
+
+    /// Load the source code located at self.path
+    ///
+    /// There are two ways to look for a source file: First in the includer's path, and
+    /// if not available in jinko's installation directory.
+    fn load(
+        &self,
+        base: &Path,
+        i: &Interpreter,
+    ) -> Result<(PathBuf, Vec<Box<dyn Instruction>>), JkError> {
+        self.load_relative(base, i)
+    }
+}
+
+impl Instruction for Incl {
+    fn kind(&self) -> InstrKind {
+        InstrKind::Statement
+    }
+
+    fn print(&self) -> String {
+        use std::ffi::OsStr;
+
+        let path: &OsStr = self.path.as_ref();
+        // FIXME: No unwrap
+        let mut base = format!("incl {}", path.to_str().unwrap());
+
+        base = match &self.alias {
+            Some(alias) => format!("{} as {}", base, alias),
+            None => base,
+        };
+
+        base
+    }
+
+    fn execute(&self, interpreter: &mut Interpreter) -> Result<InstrKind, JkError> {
+        interpreter.debug("INCL ENTER", &format!("{}", self.print()));
+
+        let base = match interpreter.path() {
+            // Get the parent directory of the interpreter's source file. We can unwrap
+            // since there's always a base
+            Some(path) => path.parent().unwrap(),
+            // The interpreter doesn't have an associated source file. Therefore, we
+            // load from where the interpreter was started. This is the case if we're
+            // in dynamic mode for example
+            None => Path::new(""),
+        };
+
+        interpreter.debug("BASE DIR", &format!("{:#?}", base));
+
+        let old_path = interpreter.path().cloned();
+
+        let (new_path, content) = self.load(base, interpreter)?;
+
+        // Temporarily change the path of the interpreter
+        interpreter.set_path(Some(new_path));
```

Comment:
Why is this needed ?

Reply:
This is needed in order for new includes to be made.

The base is computed from the source's path. 
Let's take an example:

|__ main.jk
|__ std/
    |__ lib.jk
    |__ option/
        |__ std.jk


```rust
// main.jk
incl std
```

```rust
// std/lib.jk
incl option
```

```rust
// std/option/lib.jk
type Option(...);
```

Now let's run `jinko main.jk`

The interpreter computes the parent directory of the source, and looks for a file to include. Here, std/lib.jk fits so we include it.
When going through that file, we see `incl option`. So we need to find a file that fits the requirements. In order to do that, we need to search in std/, not in the parent directory of main.jk

So we need to set the new path of the source before executing its code. If not, then the different source files that that source includes will never be found.
Then, when we're done going through that source file, we set the path back to its original source, so we can keep including from here.

Reply:
Ok but if there is an error line 177, the interpreter will return with an inconsistent state: line 180 will not have been executed

Reply:
> Ok but if there is an error line 177, the interpreter will return with an inconsistent state: line 180 will not have been executed

Yes, we need to take care of this when we implement the `SpaceLocation` struct. For example, it could keep a tiny reference to a path in order to give errors regarding to the source file the line has been found in, not the source file currently in the interpreter